source/jormungandr/jormungandr/timezone.py

```diff
@@ -61,20 +62,22 @@ def set_request_instance_timezone(instance):
         logger.warn("impossible to find timezone: '{}' for region {}".format(instance.timezone, instance.name))
 
     try:
-        g.timezone = tz
+        request.timezone = tz
+
     except RuntimeError:
         pass#working outside of an application context...
 
 
-def get_timezone():
+@functools32.lru_cache(1)
+def get_timezone(req_id):
```

Comment:
I expect this to not work well with https://github.com/CanalTP/navitia/blob/dev/source/jormungandr/jormungandr/interfaces/v1/Journeys.py#L563